app/results.js

```diff
@@ -0,0 +1,46 @@
+import { PartTypes } from './decode/morse'
+import Typography from '@mui/material/Typography'
+import Placeholder from '../component/Placeholder'
+import * as React from 'react'
+import ResultBox from '../component/ResultBox'
+
+export function decodedToResultBox(decodedVariant, key) {
+  let forReact = decodedVariant.decoded
+    .map((part, partIdx) => {
+      let string = ''
+      let color = ''
+      if (part.type === PartTypes.separator) {
+        if (part.string.length === 2) {
+          string = ' '
+        } else if (part.string.length > 2) {
+          string = '. ' + '␣'.repeat(part.string.length - 3)
+        }
+      } else if (part.type === PartTypes.char) {
+        string = part.char
+      } else {
+        string = part.string || part.char
+        color = 'warning.main'
+      }
+      return string ? (
+        <Typography key={partIdx} sx={{ color }} display="inline">
+          {string}
+        </Typography>
+      ) : null
+    })
+    .filter((part) => !!part)
+  forReact = forReact && forReact.length ? forReact : <Placeholder />
+  const resultBox = (
+    <ResultBox key={key} label={decodedVariant.label} message={forReact} />
+  )
+  return {
+    ...decodedVariant,
+    forReact: forReact,
+    resultBox: resultBox,
+  }
+}
+
+export function getResultBoxes(variantArray) {
```

Comment:
I would like to have main result also in variantArray. Then we can remove calls of decodedToResultBox in braille/morse/semaphore